helm-charts/common/speecht5/templates/deployment.yaml

```diff
@@ -27,6 +27,32 @@ spec:
       {{- end }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if not (hasPrefix "/data/" .Values.TTS_MODEL_PATH) }}
+      initContainers:
+        - name: model-downloader
+          envFrom:
+            - configMapRef:
+                name: {{ include "speecht5.fullname" . }}-config
+          securityContext:
+            {{- if .Values.global.modelUseHostPath }}
+            {}
+            {{- else }}
+            {{- toYaml .Values.securityContext | nindent 12 }}
+            {{- end }}
+          image: huggingface/downloader:0.17.3
+          command: ['sh', '-c']
+          args:
+            - |
+              huggingface-cli download --cache-dir /data --token $(HF_TOKEN)  {{ .Values.TTS_MODEL_PATH | quote }};
+              huggingface-cli download --cache-dir /data --token $(HF_TOKEN)  {{ .Values.VOCODER_MODEL| quote }};
+              {{- if .Values.global.modelUseHostPath }}
+              chown -R 1000 /data/models--{{ replace "/" "--" .Values.TTS_MODEL_PATH }};
+              chown -R 1000 /data/models--{{ replace "/" "--" .Values.VOCODER_MODEL }}
+              {{- end }}
```

Comment:
Here we want to make sure the initcontainer and container pod run with the same userID, hence the downloaded model files and the process using them should belong to the same user to avoid various permission issue.  When global.modelUseHostPath is set, no securityContext will be apllied so the container pod will run with the default userID of the image. The default userID of tei/tgi/vllm/downloader is root(0), while the default userID of speecht5 and whisper is 1000. The chown 1000 is not from the user defined securityContext, but is from the speecht5/whisper image's dockerfile.

Reply:
> Note: In `vllm` the user ID is commented out, but that's probably because of the issue this PR is fixing. If TGI works as user 1000, vLLM should also work.

Last time I remembered, vllm can not run with userID 1000. @yongfengdu may have more insights on this

Reply:
> Here we want to make sure the initcontainer and container pod run with the same userID, hence the downloaded model files and the process using them should belong to the same user to avoid various permission issue.

Yes, that was my point.  Why the same `chmod` is not done for all the other containers running as user ID 1000?

> The chown 1000 is not from the user defined securityContext, but is from the speecht5/whisper image's dockerfile.

As you can see from above `grep` output, `values.yaml` files in all of them (except vllm) include `securityContext`, not just speecht5/whisper, and pod container `securityContext` _overrides_ anything set in container Dockerfile.

Note: to test this PR with `modelUseHostPath`, you should be using a node which has no models, not one where those have been already downloaded.

Reply:
> Yes, that was my point. Why the same chmod is not done for all the other containers running as user ID 1000?
> As you can see from above `grep` output, `values.yaml` files in all of them (except vllm) include `securityContext`, not just speecht5/whisper, and pod container `securityContext` _overrides_ anything set in container Dockerfile.

If `modelUseHostPath` is set, both the initContainer and normal container will run with the same securityContext, hence the newly downloaded model files will owned by the same userID, so there is no need to do `chown`.

If `modelUseHostPath` is not set, both initContainer and normal container will run with the default userID defined in its corresponding Dockerfile respectively. The default userID of the container images are listed as below:
   - huggingface/downloader: 0
   - tgi, tei, vllm: 0
   - whisper, speecht5: 1000
 That's why we need `chown` for whisper and speecht5 cases to make sure the model files owners are the same, because huggingface/downloader image has a different default userID with whipser and speecht5 images.

Reply:
> If modelUseHostPath is NOT set, both initContainer and normal container will run without securityContext, 

Oh... I did not realize it was that broken security-wise.  You should fix that at the same time.

If non-root user works for PVC, it should work also for host FS, when data is fixed to have the correct user ID.

Reply:
done. securityContext is always applied to the main container